python/eos/analysis_file.py

```diff
@@ -425,7 +277,24 @@ def validate(self):
                     known_params[param]
                 except RuntimeError:
                     messages.append(f"Error in prediction {p_name}: Fixed parameter '{param}' not known to EOS")
-
+        for step_id, step in self._steps.items():
+            # Check that all the dependencies of a step exist
+            if (unknown_dependencies := step.depends_on - self._steps.keys()):
+                raise ValueError(f'Step \'{step_id}\' depends on unknown steps: {unknown_dependencies}')
+            # Check that the default arguments correspond to real tasks
+            if (unknown_tasks := step.default_arguments.keys() - eos.tasks._tasks.keys()):
+                raise ValueError(f'Step \'{step_id}\' has default arguments for unknown tasks: {unknown_tasks}')
+            for tc in step.tasks:
+                # Check all posteriors named in steps exist
+                if 'posterior' in tc.arguments:
+                    if tc.arguments['posterior'] not in self._posteriors:
+                        messages.append(f"Error in step {step_id}: Posterior '{task.arguments['posterior']}' not known to EOS")
+                # Check that default arguments in steps can be applied to all specified tasks
+                task_func = eos.tasks._tasks[tc.task]
```

Comment:
You are not yet checking if ``tc.task`` is a valid task.

Reply:
That is enforced by the `__post_init__` of `TaskComponent`, so must be true once we have loaded the analysis file successfully.